core/templates/pages/missing-translations-handler.spec.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+/**
+ * @fileoverview Unit tests for the Missing Translations Handler.
+ */
+
+import { AppConstants } from 'app.constants';
+import { MyMissingTranslationHandler } from './missing-translations-handler';
+
+describe('Missing Translations Handler', () => {
+  let mth = new MyMissingTranslationHandler();
+
+  it ('should create handle default translations', () => {
+    let value = 'Library';
+    spyOn(mth, 'getAppConstants').and.returnValue(
+      {DEFAULT_TRANSLATIONS: {
+        key: value
+      }} as unknown as typeof AppConstants);
```

Comment:
getAppConstants expects the return type to be of type AppConstants. So, I have to explicitly typecast this.
If I don't use unknown here, then there will be a typescript linting error telling, `this type cannot be typecasted to that`.

usage of `unknown` helps to avoid the linting issue.

Reply:
![Uploading image.png…]()